src/Day1.hs

```diff
@@ -114,12 +113,13 @@ update input st
   -- FOR DEUGGING ABOVE
   | otherwise = case scene st of
       Scene'Start -> return $ updateStart input st
-      Scene'Play -> return $ st { playState = updatePlayState input (playState st) }
+      Scene'Play -> return $ if hp <= 0 then st { scene = Scene'GameOver } else st { playState = updatePlayerHp $ updatePlayState input (playState st) }
```

Comment:
It's clearer to separate this logic into another function. eg. `updatePlayState`. (leaky abstraction) Though it works.